File: backend/queute/src/main/java/tn/cloudnine/queute/model/roadmap/StepProgress.java
package tn.cloudnine.queute.model.roadmap;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class StepProgress {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id ;
    private boolean isDone;
    private String spDescription;
    private String spRequiredPaper ;
    private Long stepId ;
}

File: backend/template/src/app/models/StepProgress.ts
export class StepProgress {
    id: number | null;
    done: boolean;
    spDescription: string;
    spRequiredPaper: string;
    stepId: number;

    constructor(init?: Partial<StepProgress>) {
        this.id = init?.id ?? null;
        this.done = init?.done ?? false;
        this.spDescription = init?.spDescription ?? "";
        this.spRequiredPaper = init?.spRequiredPaper ?? "";
        this.stepId = init?.stepId ?? 0;
    }
}